### ISSUE TITLE
nvim process crash when I do nvr 

### ISSUE THREAD

**Issue by eyalk5** [neovim/neovim#12165]
- `nvim --version`: v0.5.0-295-g234232ff4 (this is the crash dump version - but it happens in v0.5.0-418 too) 
- Operating system/version: MACOS 
- Terminal name/version: goneovim 

### Actual behaviour
Hi,

Neovim's process crashes when I run the following script:

/Users/eyalkarni/.pyenv/shims/nvr --nostart --servername `cat /tmp/curserver` --remote-send "$1GzOzz"

where $1 contains the line number.

I run this script when I do what is known as a backward search from skim. It is triggered by double click. If I do double click too fast, neovim process crashes - both in goneovim and in neovim-qt , but not in the original nvim. Yet, the crashing process is nvim. 

Now, it doesn't happen when I tried to run it in a loop in sh. Just when I double click the mouse to trigger it and just sometimes. So, I wasn't able to reproduce it in a comfortable way. But it happens a lot. 
This is nightly version. I wanted to generate a better crash dump , but couldn't compile neovim. 
It didn't work. Tell me what details you want, the crash dump is 1.35 gb. 

```
Core file '/cores/core.38795' (x86_64) was loaded.
(lldb) bt all
* thread #1, stop reason = signal SIGSTOP
  * frame #0: 0x00007fff7ebbab66 libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007fff7ed85080 libsystem_pthread.dylib`pthread_kill + 333
    frame #2: 0x00007fff7eb161ae libsystem_c.dylib`abort + 127
    frame #3: 0x000000010411be5a nvim`rpc_set_client_info + 410
    frame #4: 0x0000000103f71374 nvim`nvim_set_client_info + 1284
    frame #5: 0x0000000103f5ac2e nvim`handle_nvim_set_client_info + 638
    frame #6: 0x000000010411d2b2 nvim`request_event + 98
    frame #7: 0x0000000104026459 nvim`multiqueue_process_events + 233
    frame #8: 0x0000000104139313 nvim`nv_event + 51
    frame #9: 0x000000010412bf8d nvim`normal_execute.llvm.3303905483080372081 + 9341
    frame #10: 0x00000001042300e9 nvim`state_enter + 345
    frame #11: 0x00000001040d5763 nvim`main + 15539
    frame #12: 0x00007fff7ea6a015 libdyld.dylib`start + 1
```
from nvim`nvim_set_client_info +1284 
```

 0x103f7136f <+1279>: callq  0x10411bcc0               ; rpc_set_client_info
    0x103f71374 <+1284>: addq   $0x20, %rsp
    0x103f71378 <+1288>: movq   0x3afca9(%rip), %rax      ; (void *)0x00007fffb754f070: __stack_chk_guard
    0x103f7137f <+1295>: movq   (%rax), %rax
    0x103f71382 <+1298>: cmpq   -0x30(%rbp), %rax

```
from nvim`rpc_set_client_info:

```
    0x10411be50 <+400>: jmp    0x103f91640               ; channel_info_changed
    0x10411be55 <+405>: callq  0x1042b2b90               ; symbol stub for: abort
    0x10411be5a <+410>: nopw   (%rax,%rax)

General Purpose Registers:
       rbx = 0x000000000000006d
       rbp = 0x00007ffeebd8aae0
       rsp = 0x00007ffeebd8aab0
       r12 = 0x00007ff560580170
       r13 = 0x00007ff55b233a00
       r14 = 0x000000000000006d
       r15 = 0x00007ff55e2fdc00
       rip = 0x000000010411be5a  nvim`rpc_set_client_info + 410
13 registers were unavailable.


```

you can see it jumps to abort. I am not sure what causes it to jump. 
```
nvim`rpc_set_client_info:
    0x10411bcc0 <+0>:   pushq  %rbp
    0x10411bcc1 <+1>:   movq   %rsp, %rbp
    0x10411bcc4 <+4>:   pushq  %r15
    0x10411bcc6 <+6>:   pushq  %r14
    0x10411bcc8 <+8>:   pushq  %r13
    0x10411bcca <+10>:  pushq  %r12
    0x10411bccc <+12>:  pushq  %rbx
    0x10411bccd <+13>:  pushq  %rax
    0x10411bcce <+14>:  leaq   0x2467a3(%rip), %rax      ; channels
    0x10411bcd5 <+21>:  movq   (%rax), %rax
    0x10411bcd8 <+24>:  movq   (%rax), %r9
    0x10411bcdb <+27>:  movl   (%r9), %r8d
    0x10411bcde <+30>:  testl  %r8d, %r8d
    0x10411bce1 <+33>:  je     0x10411bd69               ; <+169>
    0x10411bce7 <+39>:  leal   -0x1(%r8), %r10d
    0x10411bceb <+43>:  movq   %rdi, %rax
    0x10411bcee <+46>:  shrq   $0x21, %rax
    0x10411bcf2 <+50>:  xorl   %edi, %eax
    0x10411bcf4 <+52>:  movl   %edi, %r14d
    0x10411bcf7 <+55>:  shll   $0xb, %r14d
    0x10411bcfb <+59>:  xorl   %eax, %r14d
    0x10411bcfe <+62>:  andl   %r10d, %r14d
    0x10411bd01 <+65>:  movq   0x10(%r9), %r11
    0x10411bd05 <+69>:  movl   %r14d, %eax
    0x10411bd08 <+72>:  shrl   $0x4, %eax
    0x10411bd0b <+75>:  movl   (%r11,%rax,4), %edx
    0x10411bd0f <+79>:  leal   (%r14,%r14), %ecx
    0x10411bd13 <+83>:  andl   $0x1e, %ecx
    0x10411bd16 <+86>:  movl   %edx, %eax
    0x10411bd18 <+88>:  shrl   %cl, %eax
    0x10411bd1a <+90>:  testb  $0x2, %al
    0x10411bd1c <+92>:  jne    0x10411bd75               ; <+181>
    0x10411bd1e <+94>:  movl   $0x1, %esi
    0x10411bd23 <+99>:  movl   %r14d, %ebx
    0x10411bd26 <+102>: nopw   %cs:(%rax,%rax)
    0x10411bd30 <+112>: testb  $0x1, %al
    0x10411bd32 <+114>: jne    0x10411bd40               ; <+128>
    0x10411bd34 <+116>: movq   0x18(%r9), %r15
    0x10411bd38 <+120>: movl   %ebx, %eax
    0x10411bd3a <+122>: cmpq   %rdi, (%r15,%rax,8)
    0x10411bd3e <+126>: je     0x10411bd78               ; <+184>
    0x10411bd40 <+128>: addl   %esi, %ebx
    0x10411bd42 <+130>: andl   %r10d, %ebx
    0x10411bd45 <+133>: cmpl   %r14d, %ebx
    0x10411bd48 <+136>: je     0x10411be55               ; <+405>
    0x10411bd4e <+142>: movl   %ebx, %eax
    0x10411bd50 <+144>: shrl   $0x4, %eax
    0x10411bd53 <+147>: movl   (%r11,%rax,4), %edx
    0x10411bd57 <+151>: leal   (%rbx,%rbx), %ecx
    0x10411bd5a <+154>: andl   $0x1e, %ecx
    0x10411bd5d <+157>: movl   %edx, %eax
    0x10411bd5f <+159>: shrl   %cl, %eax
    0x10411bd61 <+161>: incl   %esi
    0x10411bd63 <+163>: testb  $0x2, %al
    0x10411bd65 <+165>: je     0x10411bd30               ; <+112>
    0x10411bd67 <+167>: jmp    0x10411bd78               ; <+184>
    0x10411bd69 <+169>: xorl   %ebx, %ebx
    0x10411bd6b <+171>: cmpl   %r8d, %ebx
    0x10411bd6e <+174>: jne    0x10411bd8e               ; <+206>
    0x10411bd70 <+176>: jmp    0x10411be55               ; <+405>
    0x10411bd75 <+181>: movl   %r14d, %ebx
    0x10411bd78 <+184>: movl   $0x3, %eax
    0x10411bd7d <+189>: shll   %cl, %eax
    0x10411bd7f <+191>: testl  %edx, %eax
    0x10411bd81 <+193>: cmovnel %r8d, %ebx
    0x10411bd85 <+197>: cmpl   %r8d, %ebx
    0x10411bd88 <+200>: je     0x10411be55               ; <+405>
    0x10411bd8e <+206>: movq   0x20(%r9), %rax
    0x10411bd92 <+210>: movl   %ebx, %ecx
    0x10411bd94 <+212>: movq   (%rax,%rcx,8), %r13
    0x10411bd98 <+216>: testq  %r13, %r13
    0x10411bd9b <+219>: je     0x10411be55               ; <+405>
    0x10411bda1 <+225>: cmpb   $0x0, 0x698(%r13)
    0x10411bda9 <+233>: je     0x10411be55               ; <+405>
    0x10411bdaf <+239>: cmpb   $0x0, 0x6a8(%r13)
    0x10411bdb7 <+247>: jne    0x10411be55               ; <+405>
    0x10411bdbd <+253>: leaq   0x6d8(%r13), %r15
    0x10411bdc4 <+260>: movq   0x6d8(%r13), %r14
    0x10411bdcb <+267>: movq   0x6e0(%r13), %rbx
    0x10411bdd2 <+274>: testq  %rbx, %rbx
    0x10411bdd5 <+277>: je     0x10411be18               ; <+344>
    0x10411bdd7 <+279>: leaq   0x10(%r14), %r12
    0x10411bddb <+283>: nopl   (%rax,%rax)
    0x10411bde0 <+288>: movq   -0x10(%r12), %rdi
    0x10411bde5 <+293>: testq  %rdi, %rdi
    0x10411bde8 <+296>: je     0x10411bdef               ; <+303>
    0x10411bdea <+298>: callq  0x1042b2ce6               ; symbol stub for: free
    0x10411bdef <+303>: movq   0x18(%r12), %rax
    0x10411bdf4 <+308>: movq   0x10(%r12), %rcx
    0x10411bdf9 <+313>: movq   (%r12), %rdx
    0x10411bdfd <+317>: movq   0x8(%r12), %rsi
    0x10411be02 <+322>: pushq  %rax
    0x10411be03 <+323>: pushq  %rcx
    0x10411be04 <+324>: pushq  %rsi
    0x10411be05 <+325>: pushq  %rdx
    0x10411be06 <+326>: callq  0x103f65a60               ; api_free_object
    0x10411be0b <+331>: addq   $0x20, %rsp
    0x10411be0f <+335>: addq   $0x30, %r12
    0x10411be13 <+339>: decq   %rbx
    0x10411be16 <+342>: jne    0x10411bde0               ; <+288>
    0x10411be18 <+344>: movq   %r14, %rdi
    0x10411be1b <+347>: callq  0x1042b2ce6               ; symbol stub for: free
    0x10411be20 <+352>: leaq   0x10(%rbp), %rax
    0x10411be24 <+356>: movq   %rax, %rcx
    0x10411be27 <+359>: movq   0x10(%rcx), %rax
    0x10411be2b <+363>: movq   %rax, 0x10(%r15)
    0x10411be2f <+367>: movq   (%rcx), %rax
    0x10411be32 <+370>: movq   0x8(%rcx), %rcx
    0x10411be36 <+374>: movq   %rcx, 0x8(%r15)
    0x10411be3a <+378>: movq   %rax, (%r15)
    0x10411be3d <+381>: xorl   %esi, %esi
    0x10411be3f <+383>: movq   %r13, %rdi
    0x10411be42 <+386>: addq   $0x8, %rsp
    0x10411be46 <+390>: popq   %rbx
    0x10411be47 <+391>: popq   %r12
    0x10411be49 <+393>: popq   %r13
    0x10411be4b <+395>: popq   %r14
    0x10411be4d <+397>: popq   %r15
    0x10411be4f <+399>: popq   %rbp
    0x10411be50 <+400>: jmp    0x103f91640               ; channel_info_changed
    0x10411be55 <+405>: callq  0x1042b2b90               ; symbol stub for: abort
    0x10411be5a <+410>: nopw   (%rax,%rax)
```

**Comment by eyalk5** [comment]
I  suggested to add symbols to nightly BTW, so it would be easier for us all.

**Comment by erw7** [comment]
Perhaps this is a known problem.

Ref. #11052, mhinz/neovim-remote#134.

**Comment by eyalk5** [comment]
Yes, this is probably the issue. I know because it was solved when I moved to pynvim 0.3.2 . Thanks a lot!
